### PR TITLE
adittrich/doc added extensive documentation

### PR DESCRIPTION
**Pull request by alexanderdittrich**
## Description

Extensive documentation have been added to pam_interfaces.
Additionally a redundant license file have been deleted and the not used command.cpp which have been wrongly in the include directory.

Files have been reformatted using black and mpi_cpp_format.

## How I Tested

Repo builds successfully with colcon in pam environment.
Previous unit tests also run successfully.

## Do not merge before

@luator I again rebased the commit history and in between i had the same issue with authored commits. I remember that we had an issue here in my last PR in pam_models. Could you check if the same issue is here or if you notice something? 

TODO check will fail, since there was a previous TODO in the code. I left the TODO in.

### CODE REVIEW

File: include/pam_interface/state/robot.hpp [hunk 1]
```diff
@@ -145,7 +203,7 @@ class RobotState
     }
 
 private:
-    void copy(const RobotState<NB_DOFS>& other);
+    // TODO: void copy(const RobotState<NB_DOFS>& other);
```
Comment (luator) [inline include/pam_interface/state/robot.hpp:206]: What is this TODO about?

**Review by luator — COMMENTED** [review]
Thanks for this extensive documentation!  I'm not so familiar with this package myself, so can't really judge correctness but commented on a few things that were not clear to me from reading the added documentation (i.e. where more information could be useful).

The commit history of the branch looks good to me.

The "C++ Formatting" check is failing for file `srcpy/wrappers.cpp` could you check this?  In case it is properly formatted for you, what version of clang-format do you have?  Unfortunately, different versions behave differently in some cases :/

File: include/pam_interface/real/pamy1/driver.hpp [hunk 2]
```diff
@@ -6,10 +6,24 @@
 
 namespace pam_interface
 {
+/**
+ * @brief Pamy1Driver child class.
+ *
+ * Encapsulates an instance of Pamy2Interface
```
[the remark this answers is not in the document]
Comment (alexanderdittrich) [reply include/pam_interface/real/pamy1/driver.hpp:12]: in this case, it is intended that the driver encapsulates the interface or i had a misunderstanding of the use of the driver.

[the remark this answers is not in the document]
Comment (luator) [reply include/pam_interface/real/pamy1/driver.hpp]: I was just wondering if the Pamy**1**Driver really uses a Pamy**2**Interface (and not a Pamy1Interface).

File: include/pam_interface/dummy/interface.hpp [hunk 3]
```diff
@@ -10,36 +10,157 @@
 
 namespace pam_interface
 {
+/**
+ * @brief Dummy Interface class
```
[the remark this answers is not in the document]
Comment (alexanderdittrich) [reply include/pam_interface/dummy/interface.hpp]: @vincentberenz i added "Dummy robot driver can be used for testing and debugging of pam_interface, if real robot environment is not available.". Please correct me, if dummy has additional or another functionality.

File: include/pam_interface/real/pamy1/driver.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderdittrich) [reply include/pam_interface/real/pamy1/driver.hpp]: Ah i have overseen the numbers. Numbers are fixed. Thank you.

**Comment by alexanderdittrich** [comment]
Thanks a lot for the comprehensive review @luator. I hopefully resolved most of the comments. 

Unfortunately, the check still fails, even after `mpi_cpp_format`. My current clang-format version is `clang-format version 10.0.0-4ubuntu1`. According to `apt` this is up-to-date.

**Comment by luator** [comment]
@alexanderdittrich I updated the formatting check, can you rebase your branch on master to see if that solves the issue?

**Comment by alexanderdittrich** [comment]
@luator worked perfectly! Thanks!

**Review by luator — APPROVED** [review]

**Comment by alexanderdittrich** [comment]
@vincentberenz can i merge with master?

**Pull request merged**